starport/pkg/cmdrunner/cmdrunner.go

```diff
@@ -70,6 +70,7 @@ func (r *Runner) Run(ctx context.Context, steps ...*step.Step) error {
 	}
 	g, ctx := errgroup.WithContext(ctx)
 	for _, s := range steps {
+		s := s
```

Comment:
Mmh I don't understand this line
Maybe you could add a comment for future developers

Reply:
> Very cool 👍
> Personally, I would prefer to keep the different command tests in separate files

Maybe we have a `tests` directory at the top level to keep everything there?

Reply:
For this line, I saw that there was a possible bug because we read an iterated _step.Step_ value in the loop [here](https://github.com/tendermint/starport/blob/19e583a7bae29f0de6d77139aba7b5d512b79dfa/starport/pkg/cmdrunner/cmdrunner.go#L95) inside a function where it is later called in a goroutine [here](https://github.com/tendermint/starport/blob/19e583a7bae29f0de6d77139aba7b5d512b79dfa/starport/pkg/cmdrunner/cmdrunner.go#L117). Gorotines may get scheduled after the loop has done iterating which may cause them to read the last iteration in the loop. similar to [this](https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable)

We definitely need tests and comments all over the `cmdrunner` package. I think this is a must to have since we heavily depend on it! :alien: In the beginning we were experimenting with this pkg to see if it answers to our needs, turns out that it's pretty useful and its API is near to be OK. I think we can start adding tests and docs quite soon. :smile: 

solved by 3c3e694

Reply:
Ok, thanks for the explanation 👍